scripts/meta/west/runner/lpc21isp.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2018 Zilogic Systems.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for flashing LPC17XX devices with lpc21isp tool.'''
+
+from os import path
+
+from .. import log
+from .core import ZephyrBinaryRunner, RunnerCaps
+
+
+class Lpc21ispBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for lpc21isp.'''
+
+    def __init__(self, cfg, port, baud=115200, flash_freq='12000', flashtool='lpc21isp'):
+        super(Lpc21ispBinaryRunner, self).__init__(cfg)
+        self.bin_name = cfg.kernel_bin
+        self.port = port
+        self.baud = baud
+        self.flash_freq = flash_freq
+        self.flashtool = flashtool
+
+    @classmethod
+    def name(cls):
+        return 'lpc21isp'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+
+        parser.add_argument('--flashtool', default='lpc21isp',
+                            help='lpc21isp')
+        parser.add_argument('--port', default='/dev/ttyUSB0',
+                            help='serial port to flash, default /dev/ttyUSB0')
+        parser.add_argument('--baud-rate', default='115200',
+                            help='serial baud rate, default 115200')
+        parser.add_argument('--flash-freq', default='12000',
+                            help='flash frequency, default "12000"')
```

Comment:
Isn't that the oscillator frequency rather than the flash frequency? Also we need to specify the units.

Reply:
fixed